### ISSUE TITLE
[Feature][Jira] Support multi platform when resolve commit url

### ISSUE THREAD

**Issue by narrowizard** [apache/incubator-devlake#4331]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar feature requirement.

### Use case

Before, we use gitlab x jira, jira plugin resolved repo_url properly in issue_repo_commits table.  
Now, we use bitbucket x jira, repo_url in issue_repo_commits all becomes empty string. (resolve failed).  

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by klesh** [comment]
## Root cause:

The commit URLs pointing to **bitbucket** are different from those from **gitlab**:

```
https://gitlab.com/merico-dev/ee/ee-frontend/-/commit/a50227e06abd28033b27b4859c2426000f7f9c6c
https://github.com/apache/incubator-devlake/commit/a58b109e8c37c6c4fe4ec0df5a42fddb174cdec8
https://bitbucket.org/narro2/nirvana/commits/5080b020be36256b5334b869cf8db505cfdee1dc
```

However, the `repo_url` extraction logic is hardcoded:
![img_v2_292f648f-d679-4282-91c6-56e3a9699e1g](https://user-images.githubusercontent.com/61080/216908771-3f30b7d5-9be3-43ef-a6fc-d33288824708.jpg)

## Problem

Obviously, the current implementation is problematic since we can count on other code-hosting platforms likely to have different URL formats other than the above. We need to find another solution so that users may specify the `repo_url` part out of the `commit_url` by `regex` (maybe) assuming the `repo_url` is part of the `commit_url`.

## Solution

1. Add another field to `JiraTransformationRule` for users to specify the `regex` for `repo_url`
2. Enhance the existing `RemotelinkCommitShaPattern` to accept `submatch`, one for `repo_url`, one for `commit_sha` either by Order or [Naming](https://stackoverflow.com/a/20751656/20763223)

**Comment by narrowizard** [comment]
A naming regexp group is more readable. Upper vote for it.

**Comment by Startrekzky** [comment]
This one is fixed by #4526
